platform/platform_mac.mm

```diff
@@ -39,80 +66,37 @@
-    m_resourcesDir = bundlePath + "/../../data/";
-    if (!IsFileExistsByFullPath(m_resourcesDir))
+    // We're the console app (tests or tools), and resource path is a running directory.
+    ForEachPossibleDir(resourcesPath, [this](std::string const & dir)
     {
-      // Check development environment without symlink but with git repo
-      std::string const repoPath = bundlePath + "/../../../omim/data/";
-      if (IsFileExistsByFullPath(repoPath))
-        m_resourcesDir = repoPath;
-      else
-        m_resourcesDir = "./data/";
-    }
-#endif // STANDALONE_APP
-    m_writableDir = m_resourcesDir;
+      m_resourcesDir = m_writableDir = dir;
+    });
   }
   else
   {
-    m_resourcesDir = resourcesPath + "/";
-    std::string const currentDir = [NSFileManager.defaultManager currentDirectoryPath].UTF8String;
-    std::string const paths[] =
+    // We are the bundled executable with its own resources.
+    m_resourcesDir = resourcesPath;
+    ForEachPossibleDir(resourcesPath + "/../../..", [this](std::string const & dir)
     {
-      // Developers can set a symlink to the data folder.
-      m_resourcesDir + "../../../data/",
-      // Check development environment without a symlink but with a git repo.
-      m_resourcesDir + "../../../../omim/data/",
-      m_resourcesDir + "../../../../organicmaps/data/",
-      // Working directory is set to the data folder or any project's subfolder.
-      currentDir + "/../data",
-      // Working directory is set to the project's root.
-      currentDir + "/data",
-      // Working directory is set to the build folder with binaries.
-      currentDir + "/../omim/data",
-      currentDir + "/../organicmaps/data",
-    };
-    // Find the writable path.
-    for (auto const & path : paths)
-    {
-      if (IsFileExistsByFullPath(path))
-      {
-        m_writableDir = path;
-        break;
-      }
-    }
-
-    if (m_writableDir.empty())
-    {
-      auto const p = m_resourcesDir.find("/omim/");
-      if (p != std::string::npos)
-        m_writableDir = m_resourcesDir.substr(0, p) + "/omim/data/";
-    }
+      m_writableDir = dir;
+    });
 
     if (m_writableDir.empty())
     {
       NSArray * dirPaths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
       NSString * supportDir = [dirPaths objectAtIndex:0];
       m_writableDir = supportDir.UTF8String;
 #ifdef BUILD_DESIGNER
-      m_writableDir += "/OMapsData.Designer/";
+      m_writableDir += "/OMapsData.Designer";
 #else // BUILD_DESIGNER
-      m_writableDir += "/OMapsData/";
+      m_writableDir += "/OMapsData";
 #endif // BUILD_DESIGNER
       ::mkdir(m_writableDir.c_str(), 0755);
     }
   }
 
-  if (m_resourcesDir.empty())
-    m_resourcesDir = ".";
-  m_resourcesDir = base::AddSlashIfNeeded(m_resourcesDir);
-  m_writableDir = base::AddSlashIfNeeded(m_writableDir);
 
   m_settingsDir = m_writableDir;
-  m_privateDir = m_writableDir;
```

Comment:
So what will happen further, when we will be at this point with empty resource/writable dir ?
ASSERT on JoinPath somewhere? So why not to put at least CHEK(!path.epmty()) ?